irohad/util/proto_status_tools.cpp

```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "util/proto_status_tools.hpp"
+
+#include <boost/assign.hpp>
+#include <boost/bimap/bimap.hpp>
+#include <boost/bimap/set_of.hpp>
+#include <optional>
+#include "common/bind.hpp"
+#include "util/status.hpp"
+
+using iroha::operator|;
+
+namespace iroha {
+  namespace utility_service {
+
+    using ProtoStatusBimap =
+        boost::bimaps::bimap<boost::bimaps::set_of<Status>,
+                             boost::bimaps::set_of<proto::Status::StatusEnum>>;
+
+    static const ProtoStatusBimap &getProtoStatusBimap() {
+      // clang-format off
+      static const ProtoStatusBimap map =
+          boost::assign::list_of<ProtoStatusBimap::relation>
+            (Status::kUnknown,        proto::Status_StatusEnum_unknown)
+            (Status::kInitialization, proto::Status_StatusEnum_initialization)
+            (Status::kRunning,        proto::Status_StatusEnum_running)
+            (Status::kTermination,    proto::Status_StatusEnum_termination)
+            (Status::kStopped,        proto::Status_StatusEnum_stopped)
+            (Status::kFailed,         proto::Status_StatusEnum_failed);
+      // clang-format on
```

Comment:
I think we don't need bimap to translate enum <-> enum.

Reply:
bimap provides single point of synchronization of enums. please propose something better.

this place does not require performance efficiency, as statuses are emitted no more tha 6 times during whole runtime per shepherd connection.

Reply:
two functoins are two places that must be synchronized by developers. this is more error-prone. I see no evil even in two BSTs, because they are hidden from the developer and are guaranteed to be consistent.

Reply:
a better solution was found: e396a78